FRs #1055/1073: Refactor Provider Identification, Host Provider, Invoice Issuer (#1139)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/appendix/entity_identification.md

Reviewer: As discussed in Jun 13 TF3, we will need to add some clarity / scenarios on how the FinOps vendor factors into this.  

For example: Datadog is a marketplace vendor that passes through the AWS marketplace, which is then consumed by StitcherAI, potentially transformed in some way, then generated and passed along downstream.  How would this surface in these proposed fields?  Would we need _another_ field?

Author: After discussion it was decided that we would kick this feature to its own PR or even its own Feature Request due the complexities of handling data provenance in metadata.

> # Entity Identification Examples

Reviewer 2: ```suggestion
# Entity Identification Examples
```

A couple of alternatives:
- Providing Business Entities
- Participating Business Entities
- Provider Roles

> The following examples show how entity identification columns and metadata differ depending on multiple factors such as purchasing mechanism, hosting transparency, marketplace and reseller scenarios.

Reviewer 2: ```suggestion
Understanding cost and usage data in billing datasets requires identifying the roles of several participating entities involved in resource or service provisioning, invoicing, and the processing, presentation, and delivery of that data. The key participating entities include:
```

> * [Service Provider](#ServiceProvider): The entity that made the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase.
> * [Invoice Issuer](#invoiceissuer): The entity responsible for invoicing for the *resources* or *services* consumed.
> * [Host Provider](#HostProvider): The entity whose [*resources*](#glossary:resource) are used by the Service Provider to make their [*resources*](#glossary:resource) or [*services*](#glossary:service) available.
> * [Data Generator](#DataGenerator): The entity that generated the [*FOCUS dataset*](#glossary:FOCUS-dataset).

Reviewer 2: Option A: Extended Model with Additional Entity

```suggestion
* [Host Provider](#HostProvider): The entity that provides the underlying infrastructure on which the [*resources*](#glossary:resource) or [*services*](#glossary:service) of one or more Service Providers are deployed.
* [Service Provider](#ServiceProvider): The entity that provides the *resources* or *services* available for usage or purchase. These services can be built on top of infrastructure provided by a Host Provider, offered as fully integrated solutions, or include complementary offerings such as support, licensing, or consulting.
* [Invoice Issuer](#invoiceissuer): The entity responsible for generating and issuing invoices based on the usage or purchase of *resources* or *services*. This entity may or may not be the same as the Service Provider or [Payment Recipient](#paymentrecipient), depending on how the billing and resale relationships are structured.
* Payment Recipient: The entity entitled to receive payment for the invoiced resources and services. While often the same as the Invoice Issuer, in some cases it may be a different party, depending on the invoicing and payment flow.
* [Data Generator](#DataGenerator): The entity responsible for producing [*FOCUS dataset*](#glossary:FOCUS-dataset), typically based on data provided by the Service Provider. The Data Generator ensures the technical accuracy and delivery of the data, while the Service Provider remains accountable for the semantic accuracy of the source data, including correct measurement of usage or purchases and valid pricing.
```

**Note:** Based on the scenarios, it seems that when referring to the **Invoice Issuer**, we are actually addressing the question of **who received the payment** rather than **who issued the invoice**. Therefore, I suggest we consider introducing an additional entity or renaming the Invoice Issuer, as the current term may not be intuitive.

Author: I don't think changing the definition of invoice issuer is necessary. Bringing in the logic of payment recipient add additional complexity that I believe is out of scope.

Reviewer 2: Resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139/files#r2263078912)

> * [Service Provider](#ServiceProvider): The entity that made the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase.
> * [Invoice Issuer](#invoiceissuer): The entity responsible for invoicing for the *resources* or *services* consumed.
> * [Host Provider](#HostProvider): The entity whose [*resources*](#glossary:resource) are used by the Service Provider to make their [*resources*](#glossary:resource) or [*services*](#glossary:service) available.
> * [Data Generator](#DataGenerator): The entity that generated the [*FOCUS dataset*](#glossary:FOCUS-dataset).

Reviewer 2: ```suggestion
* [Host Provider](#HostProvider): The entity that provides the underlying infrastructure on which the [*resources*](#glossary:resource) or [*services*](#glossary:service) of one or more Service Providers are deployed.
* [Service Provider](#ServiceProvider): The entity that provides the *resources* or *services* available for usage or purchase. These services can be built on top of infrastructure provided by a Host Provider, offered as fully integrated solutions, or include complementary offerings such as support, licensing, or consulting.
* [Invoice Issuer](#invoiceissuer): The entity responsible for generating and issuing payable invoices based on the usage or purchase of *resources* or *services*. This entity may or may not be the same as the Service Provider, depending on how the billing and resale relationships are structured.
* [Data Generator](#DataGenerator): The entity responsible for producing [*FOCUS dataset*](#glossary:FOCUS-dataset), typically based on data provided by the Service Provider. The Data Generator ensures the technical accuracy and delivery of the data, while the Service Provider remains accountable for the semantic accuracy of the source data, including correct measurement of usage or purchases and valid pricing.
```

This alternative proposal includes:

- A bit more context on role responsibilities and relationships
  - Discussion topic: FinOps Vendor as DataGenerator
- An ordering that mirrors how the roles interact in practice, making the definitions easier to follow in multi-party scenarios.

**Additional note:** Based on the scenarios, it seems that when referring to the Invoice Issuer, we are actually addressing the question of who received the payment rather than who issued the invoice. We should ensure this is clear in the definition itself, without relying on the reader to infer it from the scenarios.

**TODO:** Once we finalize the definitions in the appendix, the definitions in the column descriptions should be updated accordingly.

Author: I don't want to change orders at this time because it complicates comparison. We can do that later.

Author: I think we need to focus on resolving the definition issues in the actual columns rather than here. Having the definitions in two places is problematic. It was included for ease of reading but is resulting in problems.

Author: @ijurica  I don't want to change order of columns right now as it making comparison difficult.

> The value for each of these may be different depending on the various purchasing scenarios for *resources* or *services*. Use cases for purchasing direct, via a Managed Service Provider (MSP), via a cloud marketplace, and from internal service offerings were considered. The table below presents a few scenarios to show how the value for each dimension may change based on the purchasing scenario.

Reviewer 2: nit:

```suggestion
The value for each of these may vary depending on how *resources* or *services* are obtained — whether directly from a Cloud Service Provider (CSP) or a SaaS provider, via a Managed Service Provider (MSP), through a cloud marketplace, or from internal service offerings. The table below provides examples that illustrate how the value for each dimension may shift depending on the method of acquisition and other contributing factors.
```

> | #   | Scenario                                                                                                                                                                                                                                          | Service Provider             | Invoice Issuer                                             | Host Provider                                              | Data Generator                                             |

Reviewer 2: Should we provide Example Scenarios with **FinOps Vendor** as a DataGenerator

Author: I am not convinced that a Finops Vendor is a datagenerator...they are something else.

Reviewer: For what it's worth, a FinOps Vendor (aka tooling vendor) _may or may not_ be a data generator.

[Here](https://www.vantage.sh/blog/focus-export) is an example of Vantage (a tooling vendor) now exporting FOCUS data.  At present, I believe they are the only tooling vendor both producing and consuming data.

Author: Yeah this is why we may need to have another entity attribute to support the use case where a vendor is ingesting focus datasets but not changing them vs a vendor that is ingesting and making modifications but those modifications are still FOCUS compliant.

> |:-----|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-----------------------------|:-----------------------------------------------------------|:-----------------------------------------------------------|:-----------------------------------------------------------|

Reviewer: As a comparison of changes from the previous version, @ijurica has helpfully assembled a [spreadsheet](https://docs.google.com/spreadsheets/d/1PWw4K3E6BT5ZVq8CrPd3TbGq1Cs59i5wvRgmFvj03Rg/edit?gid=545225056#gid=545225056) for us to use!

> | 1.1 | Purchasing cloud services directly from cloud provider                                                                                                                                                                                            | Cloud service provider       | Cloud service provider                                     | Cloud service provider                                     | Cloud service provider                                     |

Reviewer 2: ```suggestion
| 1.1 | Cloud resources or services acquired directly from a CSP.                                                                                                                                                                                            | CSP       | CSP                                     | CSP                                     | CSP                                     |
```

> | 1.2 | Purchasing cloud services from the cloud provider where the cloud region is operated by a 3rd party, where the cost is payable to the 3rd party                                                                                                   | Cloud service provider       | Entity operating the region for the cloud service provider | Entity operating the region for the cloud service provider | Entity operating the region for the cloud service provider |

Reviewer 2: ```suggestion
| 1.2 | Cloud resources or services acquired from a CSP, where the cloud region is operated by and payable to a 3rd party.                                                                                                   | CSP       | Entity operating the region for the cloud service provider | Entity operating the region for the cloud service provider | Entity operating the region for the cloud service provider |
```

Aren’t these charge records included in the dataset provided by the CSP (i.e.,  isn't CSP the Data Generator)

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 2.1 | Purchasing cloud services via MSP, MSP DOES provide insight into or allows customers to see hosting provider                                                                                                                                      | Managed Service Provider     | Managed Service Provider                                   | Cloud service provider                                     | Managed Service Provider                                   |

Reviewer 2: ```suggestion
| 2.1 | Cloud resources or services acquired via an MSP, with visibility into the underlying hosting provider.                                                                                                                                      | MSP     | MSP                                   | CSP                                     | MSP                                   |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 2.2 | Purchasing cloud services via MSP, MSP DOES NOT provide insight into or allows customers to see hosting provider                                                                                                                                  | Managed Service Provider     | Managed Service Provider                                   | Managed Service Provider                                   | Managed Service Provider                                   |

Reviewer 2: ```suggestion
| 2.2 | Cloud resources or services acquired via an MSP, without visibility into the underlying hosting provider.                                                                                                                                  | MSP     | MSP                                   | MSP                                   | MSP                                   |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 2.3 | Purchasing cloud-agnostic resources or services built/sold by an MSP                                                                                                                                                                              | Managed Service Provider     | Managed Service Provider                                   | Managed Service Provider                                   | Managed Service Provider                                   |

Reviewer 2: ```suggestion
| 2.3 | Cloud-agnostic resources or services acquired from an MSP                                                                                                                                                                              | MSP     | MSP                                   | MSP                                   | MSP                                   |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 2.4 | Purchasing labor services from managed service provider                                                                                                                                                                                           | Managed Service Provider     | Managed Service Provider                                   | N/A                                                        | Managed Service Provider                                   |

Reviewer 2: ```suggestion
| 2.4 | Labor services acquired from an MSP                                                                                                                                                                                           | MSP     | MSP                                   | N/A                                                        | MSP                                   |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 3.1 | Purchasing a cloud marketplace offering that runs on the cloud provider                                                                                                                                                                           | Marketplace seller           | Cloud service provider                                     | Cloud service provider                                     | Cloud service provider                                     |

Reviewer 2: ```suggestion
| 3.1 | Cloud marketplace offerings hosted in Public Cloud.                                                                                                                                                                           | Marketplace seller           | CSP                                     | CSP                                     | CSP                                     |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 3.2 | Purchase record for a cloud marketplace offering that is not running directly on your cloud infrastructure (e.g,. SaaS product, Professional Services), marketplace seller DOES provide view/selection on where resources are running             | Marketplace seller           | Cloud service provider                                     | Cloud service provider                                     | Cloud service Provider                                     |

Reviewer 2: ```suggestion
| 3.2 | Purchase records for cloud marketplace offerings (not hosted on your cloud infrastructure), with visibility into the underlying hosting provider             | Marketplace seller           | CSP                                     | CSP                                     | CSP                                     |
```

In several scenario descriptions, we mention 'not running directly on your cloud infrastructure'. 

But in this context, does it even matter whether it's your infrastructure? 
After all, this isn’t about the cost of infrastructure.

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 3.3 | Usage records for a purchased a cloud marketplace offering that is not running directly on your cloud infrastructure (e.g,. SaaS product, Professional Services), marketplace seller DOES provide view/selection on where resources are running   | Marketplace seller           | Cloud service provider                                     | Cloud service provider                                     | Marketplace Seller                                         |

Reviewer 2: ```suggestion
| 3.3 | Usage records for cloud marketplace offerings (not hosted on your cloud infrastructure), with visibility into the underlying hosting provider   | Marketplace seller           | CSP                                     | CSP                                     | Marketplace Seller                                         |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 3.4 | Purchase record for a cloud marketplace offering that is not running directly on your cloud infrastructure (e.g,. SaaS product, Professional Services), marketplace seller DOES NOT provide view/selection on where resources are running         | Marketplace seller           | Cloud service provider                                     | Marketplace seller                                         | Cloud service Provider                                     |

Reviewer 2: ```suggestion
| 3.4 | Purchase records for cloud marketplace offerings (not hosted on your cloud infrastructure), without visibility into the underlying hosting provider        | Marketplace seller           | CSP                                     | Marketplace seller                                         | CSP                                     |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 3.5 | Usage records for a purchased cloud marketplace offering that is not running directly on your cloud infrastructure (e.g,. SaaS product, Professional Services), marketplace seller DOES NOT provide view/selection on where resources are running | Marketplace seller           | Cloud service provider                                     | Marketplace seller                                         | Marketplace Seller                                         |

Reviewer 2: ```suggestion
| 3.5 | Usage records for cloud marketplace offerings (not hosted on your cloud infrastructure), without visibility into the underlying hosting provider | Marketplace seller           | CSP                                     | Marketplace seller                                         | Marketplace Seller                                         |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 3.6 | Purchasing a SaaS product that is not directly run on your cloud infrastructure from a 3rd party reseller managed cloud marketplace                                                                                                               | Reseller                     | Reseller                                                   | Reseller                                                   | Reseller                                                   |

Reviewer 2: ```suggestion
| 3.6 | SaaS software (not hosted on your cloud infrastructure) acquired via a 3rd-party reseller-operated cloud marketplace                                                                                                     | Reseller                     | Reseller                                                   | Reseller                                                   | Reseller                                                   |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 4.1 | Purchasing SaaS software directly                                                                                                                                                                                                                 | SaaS provider                | SaaS provider                                              | N/A                                                        | SaaS provider                                              |

Reviewer 2: ```suggestion
| 4.1 | SaaS software acquired directly from a SaaS provider                                                                                                                                                                                                                 | SaaS provider                | SaaS provider                                              | N/A                                                        | SaaS provider                                              |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 4.2 | Purchasing software directly from a SaaS software provider that runs on your cloud resources (Usage Records From CSP FOCUS Dataset)                                                                                                               | SaaS provider                | SaaS provider                                              | N/A                                                        | SaaS provider                                              |

Reviewer 2: ```suggestion
| 4.2 | SaaS software (hosted on your cloud infrastructure) acquired directly from a SaaS provider                                                                                                               | SaaS provider                | SaaS provider                                              | N/A                                                        | SaaS provider                                              |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 4.2 | Your Cloud service provider billing data for resources running software purchased from SaaS vendord (Usage Record From SaaS vendors FOCUS Dataset)                                                                                                | Cloud service provider       | Cloud service provider                                     | Cloud service provider                                     | Cloud service provider                                     |

Reviewer 2: ```suggestion
| 4.3 | Cloud resources acquired from a CSP, hosting software separately acquired from a SaaS provider                                                                                                | CSP       | CSP                                     | CSP                                     | CSP                                     |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 5.1 | Purchasing internal infrastructure or services offerings running on-premise                                                                                                                                                                       | Internal Name                | Internal Name                                              | Internal name                                              | Internal Name                                              |

Reviewer 2: ```suggestion
| 5.1 | Internal infrastructure or services hosted on-premise                                                                                                                                                                       | Internal Name                | Internal Name                                              | Internal name                                              | Internal Name                                              |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 5.2 | Purchasing internal infrastructure or services offerings running on cloud                                                                                                                                                                         | Internal Name                | Internal Name                                              | Cloud service provider                                     | Internal Name                                              |

Reviewer 2: ```suggestion
| 5.2 | Internal infrastructure or services hosted in Public Cloud                                                                                                                                                                         | Internal Name                | Internal Name                                              | CSP                                     | Internal Name                                              |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 5.3 | Associated software license cost for use on an on-premise infrastructure platform (Where license cost is presented separately in cost data)                                                                                                       | Licensable Software provider | License Seller                                             | N/A                                                        | License Seller                                             |

Reviewer 2: ```suggestion
| 5.3 | Software license costs for use on on-premise infrastructure, reported separately from the underlying infrastructure costs                                                                                                       | Software provider | License Seller                                             | N/A                                                        | License Seller                                             |
```

Reviewer 2: resolving in favor of [this thread](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/1139#discussion_r2262919736)

> | 5.3 | Associated software license cost for use on an on-premise infrastructure platform (Where license cost is presented separately in cost data)                                                                                                       | Licensable Software provider | License Seller                                             | N/A                                                        | License Seller                                             |

Reviewer 2: ```suggestion
| 5.3 | Associated software license cost for use on an on-premise infrastructure platform (Where license cost is presented separately in cost data)                                                                                                       | Licensable Software provider | License Seller                                             | N/A                                                        | License Seller                                             |
```

Is this limited to 'for use on an on-premise infrastructure,' or is it more broadly applicable? Considering BYOL scenarios, perhaps we can omit the 'for use on an on-premise infrastructure' qualifier?

> | 5.3 | Associated software license cost for use on an on-premise infrastructure platform (Where license cost is presented separately in cost data)                                                                                                       | Licensable Software provider | License Seller                                             | N/A                                                        | License Seller                                             |

Reviewer 2: ```suggestion
| #     | Scenario                                                                                                                                            | Host Provider                            | Service Provider             | Invoice Issuer                   | Data Generator     |
|-------|-----------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------|------------------------------|----------------------------------|--------------------|
| 1.1   | Purchasing cloud resources or services directly from a CSP                                                                                          | CSP                                      | CSP                          | CSP                              | CSP                |
| 1.2   | Purchasing cloud resources or services from a CSP, where the underlying resources are operated by a 3rd party.                                      | Entity operating the region for the CSP  | CSP                          | CSP                              | CSP                |
| 2.1   | Purchasing cloud resources or services via an MSP, with visibility into the underlying hosting provider.                                            | CSP                                      | MSP                          | MSP                              | MSP                |
| 2.2   | Purchasing cloud resources or services via an MSP, without visibility into the underlying hosting provider.                                         | MSP                                      | MSP                          | MSP                              | MSP                |
| 2.3   | Purchasing cloud-agnostic resources or services from an MSP                                                                                         | MSP                                      | MSP                          | MSP                              | MSP                |
| 2.4   | Purchasing labor services from an MSP                                                                                                               | N/A                                      | MSP                          | MSP                              | MSP                |
| 3.1.1 | Purchase records for cloud marketplace offerings running on a CSP infrastructure.                                                                   | CSP                                      | Marketplace seller           | CSP                              | CSP                |
| 3.1.2 | Usage records for cloud marketplace offerings running on a CSP infrastructure.                                                                      | CSP                                      | Marketplace seller           | CSP                              | Marketplace seller |
| 3.2.1 | Purchase records for cloud marketplace offerings not running on your cloud infrastructure, with visibility into the underlying hosting provider.    | CSP                                      | Marketplace seller           | CSP                              | CSP                |
| 3.2.2 | Usage records for cloud marketplace offerings not running on your cloud infrastructure, with visibility into the underlying hosting provider.       | CSP                                      | Marketplace seller           | CSP                              | Marketplace Seller |
| 3.3.1 | Purchase records for cloud marketplace offerings not running on your cloud infrastructure, without visibility into the underlying hosting provider. | Marketplace seller                       | Marketplace seller           | CSP                              | CSP                |
| 3.3.2 | Usage records for cloud marketplace offerings not running on your cloud infrastructure, without visibility into the underlying hosting provider.    | Marketplace seller                       | Marketplace seller           | CSP                              | Marketplace Seller |
| 3.4.1 | Purchase records for SaaS products not running on your cloud infrastructure, purchased via a reseller.                                              | Reseller                                 | Reseller                     | Reseller                         | Reseller           |
| 3.4.2 | Usage records for SaaS products not running on your cloud infrastructure, purchased via a reseller.                                                 | Reseller                                 | Reseller                     | Reseller                         | Reseller           |
| 4.1   | Purchasing SaaS products directly from a SaaS provider, with visibility into the underlying hosting provider.                                       | CSP                                      | SaaS provider                | SaaS provider                    | SaaS provider      |
| 4.2   | Purchasing SaaS products directly from a SaaS provider, without visibility into the underlying hosting provider.                                    | N/A                                      | SaaS provider                | SaaS provider                    | SaaS provider      |
| 4.3.1 | Purchasing SaaS products running on your cloud infrastructure, purchased directly from a SaaS provider (see 4.3.2 for charges related to the underlying cloud infrastructure). | CSP           | SaaS provider                | SaaS provider                    | SaaS provider      |
| 4.3.2 | Purchasing resources and services from a CSP used to host SaaS products separately acquired from a SaaS provider (see 4.3.1 for charges related to the SaaS products). | CSP                   | CSP                          | CSP                              | CSP                |
| 5.1   | Purchasing internal resources or services hosted in Data Center                                                                                     | Internal name                            | Internal Name                | Internal Name                    | Internal Name      |
| 5.2   | Purchasing Internal resources or services hosted in Public Cloud.                                                                                   | CSP                                      | Internal Name                | Internal Name                    | Internal Name      |
| 6.1   | Software license costs, reported separately from the costs of the resources or services they apply to.                                              | N/A                                      | Licensable Software provider | License Seller                   | License Seller     |
```

### FOCUS 1.3 Original proposal vs Alternative preview

| Proposal Version | #     | Scenario                                                                                                                                                                                                                                          | Host Provider                                              | Service Provider             | Invoice Issuer                                             | Data Generator                                             |
|------------------|-------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------|------------------------------|------------------------------------------------------------|------------------------------------------------------------|
| Initial          | 1.1   | Purchasing cloud services directly from cloud provider                                                                                                                                                                                            | Cloud service provider                                     | Cloud service provider       | Cloud service provider                                     | Cloud service provider                                     |
| Alternative      | 1.1   | Purchasing cloud resources or services directly from a CSP                                                                                                                                                                                        | CSP                                                        | CSP                          | CSP                                                        | CSP                                                        |
| Initial          | 1.2   | Purchasing cloud services from the cloud provider where the cloud region is operated by a 3rd party, where the cost is payable to the 3rd party                                                                                                   | Entity operating the region for the cloud service provider | Cloud service provider       | Entity operating the region for the cloud service provider | Entity operating the region for the cloud service provider |
| Alternative      | 1.2   | Purchasing cloud resources or services from a CSP, where the underlying resources are operated by a 3rd party.                                                                                                                                    | Entity operating the region for the CSP                    | CSP                          | CSP                                                        | CSP                                                        |
| Initial          | 2.1   | Purchasing cloud services via MSP, MSP DOES provide insight into or allows customers to see hosting provider                                                                                                                                      | Cloud service provider                                     | Managed Service Provider     | Managed Service Provider                                   | Managed Service Provider                                   |
| Alternative      | 2.1   | Purchasing cloud resources or services via an MSP, with visibility into the underlying hosting provider.                                                                                                                                          | CSP                                                        | MSP                          | MSP                                                        | MSP                                                        |
| Initial          | 2.2   | Purchasing cloud services via MSP, MSP DOES NOT provide insight into or allows customers to see hosting provider                                                                                                                                  | Managed Service Provider                                   | Managed Service Provider     | Managed Service Provider                                   | Managed Service Provider                                   |
| Alternative      | 2.2   | Purchasing cloud resources or services via an MSP, without visibility into the underlying hosting provider.                                                                                                                                       | MSP                                                        | MSP                          | MSP                                                        | MSP                                                        |
| Initial          | 2.3   | Purchasing cloud-agnostic resources or services built/sold by an MSP                                                                                                                                                                              | Managed Service Provider                                   | Managed Service Provider     | Managed Service Provider                                   | Managed Service Provider                                   |
| Alternative      | 2.3   | Purchasing cloud-agnostic resources or services from an MSP                                                                                                                                                                                       | MSP                                                        | MSP                          | MSP                                                        | MSP                                                        |
| Initial          | 2.4   | Purchasing labor services from managed service provider                                                                                                                                                                                           | N/A                                                        | Managed Service Provider     | Managed Service Provider                                   | Managed Service Provider                                   |
| Alternative      | 2.4   | Purchasing labor services from an MSP                                                                                                                                                                                                             | N/A                                                        | MSP                          | MSP                                                        | MSP                                                        |
| Initial          | 3.1   | Purchasing a cloud marketplace offering that runs on the cloud provider                                                                                                                                                                           | Cloud service provider                                     | Marketplace seller           | Cloud service provider                                     | Cloud service provider                                     |
| Alternative      | 3.1.1 | Purchase records for cloud marketplace offerings running on a CSP infrastructure.                                                                                                                                                                 | CSP                                                        | Marketplace seller           | CSP                                                        | CSP                                                        |
| Alternative      | 3.1.2 | Usage records for cloud marketplace offerings running on a CSP infrastructure.                                                                                                                                                                    | CSP                                                        | Marketplace seller           | CSP                                                        | Marketplace seller                                         |
| Initial          | 3.2   | Purchase record for a cloud marketplace offering that is not running directly on your cloud infrastructure (e.g,. SaaS product, Professional Services), marketplace seller DOES provide view/selection on where resources are running             | Cloud service provider                                     | Marketplace seller           | Cloud service provider                                     | Cloud service Provider                                     |
| Alternative      | 3.2.1 | Purchase records for cloud marketplace offerings not running on your cloud infrastructure, with visibility into the underlying hosting provider.                                                                                                  | CSP                                                        | Marketplace seller           | CSP                                                        | CSP                                                        |
| Initial          | 3.3   | Usage records for a purchased a cloud marketplace offering that is not running directly on your cloud infrastructure (e.g,. SaaS product, Professional Services), marketplace seller DOES provide view/selection on where resources are running   | Cloud service provider                                     | Marketplace seller           | Cloud service provider                                     | Marketplace Seller                                         |
| Alternative      | 3.2.2 | Usage records for cloud marketplace offerings not running on your cloud infrastructure, with visibility into the underlying hosting provider.                                                                                                     | CSP                                                        | Marketplace seller           | CSP                                                        | Marketplace Seller                                         |
| Initial          | 3.4   | Purchase record for a cloud marketplace offering that is not running directly on your cloud infrastructure (e.g,. SaaS product, Professional Services), marketplace seller DOES NOT provide view/selection on where resources are running         | Marketplace seller                                         | Marketplace seller           | Cloud service provider                                     | Cloud service Provider                                     |
| Alternative      | 3.3.1 | Purchase records for cloud marketplace offerings not running on your cloud infrastructure, without visibility into the underlying hosting provider.                                                                                               | Marketplace seller                                         | Marketplace seller           | CSP                                                        | CSP                                                        |
| Initial          | 3.5   | Usage records for a purchased cloud marketplace offering that is not running directly on your cloud infrastructure (e.g,. SaaS product, Professional Services), marketplace seller DOES NOT provide view/selection on where resources are running | Marketplace seller                                         | Marketplace seller           | Cloud service provider                                     | Marketplace Seller                                         |
| Alternative      | 3.3.2 | Usage records for cloud marketplace offerings not running on your cloud infrastructure, without visibility into the underlying hosting provider.                                                                                                  | Marketplace seller                                         | Marketplace seller           | CSP                                                        | Marketplace Seller                                         |
| Initial          | 3.6   | Purchasing a SaaS product that is not directly run on your cloud infrastructure from a 3rd party reseller managed cloud marketplace                                                                                                               | Reseller                                                   | Reseller                     | Reseller                                                   | Reseller                                                   |
| Alternative      | 3.4.1 | Purchase records for SaaS products not running on your cloud infrastructure, purchased via a reseller.                                                                                                                                            | Reseller                                                   | Reseller                     | Reseller                                                   | Reseller                                                   |
| Alternative      | 3.4.2 | Usage records for SaaS products not running on your cloud infrastructure, purchased via a reseller.                                                                                                                                               | Reseller                                                   | Reseller                     | Reseller                                                   | Reseller                                                   |
| Initial          | 4.1   | Purchasing SaaS software directly                                                                                                                                                                                                                 | N/A                                                        | SaaS provider                | SaaS provider                                              | SaaS provider                                              |
| Alternative      | 4.1   | Purchasing SaaS products directly from a SaaS provider, with visibility into the underlying hosting provider.                                                                                                                                     | CSP                                                        | SaaS provider                | SaaS provider                                              | SaaS provider                                              |
| Alternative      | 4.2   | Purchasing SaaS products directly from a SaaS provider, without visibility into the underlying hosting provider.                                                                                                                                  | N/A                                                        | SaaS provider                | SaaS provider                                              | SaaS provider                                              |
| Initial          | 4.2   | Purchasing software directly from a SaaS software provider that runs on your cloud resources (Usage Records From CSP FOCUS Dataset)                                                                                                               | N/A                                                        | SaaS provider                | SaaS provider                                              | SaaS provider                                              |
| Alternative      | 4.3.1 | Purchasing SaaS products running on your cloud infrastructure, purchased directly from a SaaS provider (see 4.3.2 for charges related to the underlying cloud infrastructure).                                                                    | CSP                                                        | SaaS provider                | SaaS provider                                              | SaaS provider                                              |
| Initial          | 4.2   | Your Cloud service provider billing data for resources running software purchased from SaaS vendord (Usage Record From SaaS vendors FOCUS Dataset)                                                                                                | Cloud service provider                                     | Cloud service provider       | Cloud service provider                                     | Cloud service provider                                     |
| Alternative      | 4.3.2 | Purchasing resources and services from a CSP used to host SaaS products separately acquired from a SaaS provider (see 4.3.1 for charges related to the SaaS products).                                                                            | CSP                                                        | CSP                          | CSP                                                        | CSP                                                        |
| Initial          | 5.1   | Purchasing internal infrastructure or services offerings running on-premise                                                                                                                                                                       | Internal name                                              | Internal Name                | Internal Name                                              | Internal Name                                              |
| Alternative      | 5.1   | Purchasing internal resources or services hosted in Data Center.                                                                                                                                                                                  | Internal name                                              | Internal Name                | Internal Name                                              | Internal Name                                              |
| Initial          | 5.2   | Purchasing internal infrastructure or services offerings running on cloud                                                                                                                                                                         | Cloud service provider                                     | Internal Name                | Internal Name                                              | Internal Name                                              |
| Alternative      | 5.2   | Purchasing Internal resources or services hosted in Public Cloud.                                                                                                                                                                                 | CSP                                                        | Internal Name                | Internal Name                                              | Internal Name                                              |
| Initial          | 5.3   | Associated software license cost for use on an on-premise infrastructure platform (Where license cost is presented separately in cost data)                                                                                                       | N/A                                                        | Licensable Software provider | License Seller                                             | License Seller                                             |
| Alternative      | 6.1   | Software license costs, reported separately from the costs of the resources or services they apply to.                                                                                                                                            | N/A                                                        | Licensable Software provider | License Seller                                             | License Seller                                             |

Author: I don't want to have order of column changes, lets save that for when we have agreement on the definitions.

# Entity Identification Examples

Understanding cost and usage data in billing datasets requires identifying the roles of several participating entities involved in resource or service provisioning, invoicing, and data generation. The FOCUS Specification includes multiple columns to identify key participating entities, these include:

* [Service Provider](#ServiceProvider)
* [Invoice Issuer](#invoiceissuer)
* [Host Provider](#HostProvider)

> * [Data Generator](#DataGenerator)

Reviewer 2: The PR doesn’t include changes to the Data Generator definition, so I’m adding this comment here...
I’d recommend expanding the definition to clarify the responsibilities of the Data Generator.

Current version of the Data Generator metadata definition:

> Human-readable name of the entity that is generating the data.

Proposed version:

> Human-readable name of the entity responsible for producing the [*FOCUS dataset*](#glossary:FOCUS-dataset), typically based on data provided by the [Service Provider](#serviceprovider). The Data Generator ensures the technical accuracy and delivery of the data, while the Service Provider remains accountable for the semantic accuracy of the source data.

***Note:** We might consider adding all these (business) entities to the glossary for clarity.*

---

On that note, “Entity” feels a bit too generic for this appendix. Maybe we could explore alternative terms such as:

- Business Entities
- Providing Business Entities
- Participating Business Entities

Author: Put in changes.

| 2.1   | Purchasing cloud resources or services via an MSP, with visibility into the underlying hosting provider.                                                                       | MSP                          | MSP            | CSP                                     | MSP                |
| 2.2   | Purchasing cloud resources or services via an MSP, without visibility into the underlying hosting provider.                                                                    | MSP                          | MSP            | MSP                                     | MSP                |
| 2.3   | Purchasing cloud-agnostic resources or services from an MSP                                                                                                                    | MSP                          | MSP            | MSP                                     | MSP                |
| 2.4   | Purchasing labor services from an MSP                                                                                                                                          | MSP                          | MSP            | N/A                                     | MSP                |

> | 3.1.1 | Purchase records for cloud marketplace offerings running on your CSP infrastructure.                                                                                           | Marketplace seller           | CSP            | CSP                                     | CSP                |

Reviewer 2: Consider using “Marketplace Offerings” (capitalized) instead of “marketplace offerings”.
If we decide to adopt this change, we should apply it consistently across all remaining occurrences in the documentation.

```suggestion
| 3.1.1 | Purchase records for cloud Marketplace offerings running on your CSP infrastructure.                                                                                           | Marketplace seller           | CSP            | CSP                                     | CSP                |
```
